core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -367,6 +368,52 @@ object SparkSubmit extends CommandLineUtils with Logging {
       }.orNull
     }
 
+    // Using a dummy http URI to check if HTTP(s) FileSystem is available, it returns true in
+    // Hadoop 2.9+, otherwise it returns false.
+    val isHttpFsAvailable = Try { FileSystem.get(Utils.resolveURI("http://foo/bar"), hadoopConf) }
```

Comment:
There is a Filesystem.getFileSystemClass function we could use here instead of calling dummy uri